core/src/main/kotlin/org/sert2521/sertain/motors/MotorController.kt

```diff
@@ -1,266 +1,50 @@
-        setTargetPosition(position.convertTo(encoder!!.ticks).value.toInt())
-    }
-
-    fun setTargetVelocity(velocity: Int) {
-        ctreMotorController.set(CtreControlMode.Velocity, velocity.toDouble())
-    }
-
-    fun <U1 : AngularUnit, U2 : ChronicUnit, V : AngularVelocityValue<U1, U2>> setTargetVelocity(velocity: V) {
-        checkNotNull(encoder) { "You must configure your encoder to use units." }
-        setTargetVelocity(velocity.convertTo(encoder!!.ticksPerSecond).value.toInt())
-    }
-
-    fun setCurrent(current: Double) {
-        ctreMotorController.set(CtreControlMode.Current, current)
-    }
-
-    fun disable() {
-        ctreMotorController.neutralOutput()
-    }
-
-    internal fun updatePidf(slot: Int, pidf: MotorPidf) {
-        with(pidf) {
-            ctreMotorController.apply {
-                config_kP(slot, kp)
-                config_kI(slot, ki)
-                config_kD(slot, kd)
-                config_kF(slot, kf)
-                config_IntegralZone(slot, integralZone)
-                configAllowableClosedloopError(slot, allowedError)
-                configMaxIntegralAccumulator(slot, maxIntegral)
-                configClosedLoopPeakOutput(slot, maxOutput)
-                configClosedLoopPeriod(slot, period)
-            }
-        }
-    }
-
-    private fun updateCurrentLimit(limit: CurrentLimit) {
-        eachTalon {
-            (ctreMotorController as CtreTalon).apply {
-                configContinuousCurrentLimit(limit.continuousLimit)
-                configPeakCurrentLimit(limit.maxLimit)
-                configPeakCurrentDuration(limit.maxDuration)
-                enableCurrentLimit(limit.enabled)
-            }
-        }
-    }
+    internal abstract fun updatePidf(slot: Int, pidf: MotorPidf)
+}
 
-    init {
-        eachMotor { ctreMotorController.setNeutralMode(ctreNeutralMode(brakeMode)) }
-        ctreMotorController.apply {
-            configClosedloopRamp(closedLoopRamp)
-            configOpenloopRamp(openLoopRamp)
-            configNominalOutputReverse(minOutputRange.start)
-            configNominalOutputForward(minOutputRange.endInclusive)
-            configPeakOutputReverse(maxOutputRange.start)
-            configPeakOutputForward(maxOutputRange.endInclusive)
-            selectProfileSlot(pidfSlot, 0)
-            pidf.toMap().forEach {
-                updatePidf(it.key, it.value)
-            }
-            updateCurrentLimit(currentLimit)
-        }
-        configure()
+fun<T: MotorId> motorController(id: T, vararg followerIds: MotorId, configure: MotorController.() -> Unit = {}): MotorController {
+    return when (id) {
+        is TalonId  -> TalonMotorController(id,  *followerIds, configure = configure)
+        is VictorId -> VictorMotorController(id, *followerIds, configure = configure)
+        else -> throw NotImplementedError("A Motor Controller is not implemented for this ID")
     }
 }
 
-// Current can only be read from talons
```

Comment:
The current should be able to be read from talon motor controllers. Now that there is a class specifically for talons, this should be typesafe.

Reply:
Alright, I'll implement this for `TalonMotorController`.